toplevel/opttoploop.mli

```diff
@@ -96,8 +100,8 @@ val max_printer_steps: int ref
 val parse_toplevel_phrase : (Lexing.lexbuf -> Parsetree.toplevel_phrase) ref
 val parse_use_file : (Lexing.lexbuf -> Parsetree.toplevel_phrase list) ref
 val print_location : formatter -> Location.t -> unit
-val print_error : formatter -> Location.error -> unit
-val print_warning : Location.t -> formatter -> Warnings.t -> unit
+val print_error : Misc.Log.t -> Location.error -> unit
+val print_warning : Location.t -> Misc.Log.t -> Warnings.t -> unit
```

Comment:
again, we would have the option to keep `print_*` unchanged and have additional `log_{error,warning}` functions or `print_{error,warning}_with_log`. Not sure which is best.